src/tailwind.scroll-shadows.js

```diff
@@ -0,0 +1,62 @@
+import plugin from 'tailwindcss/plugin.js';
+
+/**
+ * Utility class to show scroll-hint shadows at the top and bottom of a
+ * scrollable frame element if:
+ *  - The content height exceeds the frame height: i.e. can be scrolled, and
+ *  - The content is scrollable in the shadow's direction (up or down)
+ *
+ * Shadows are not visible once the frame has been scrolled all the way in the
+ * shadow's direction. Shadows are not visible if the height of the content
+ * does not overflow the frame (is not scrollable).
+ *
+ * The shadow hinting is handled by four positioned background gradients:
+ *   - One gradient each at top and bottom of frame that obscure the shadow hints
+ *     (shadow covers). These use `background-attachment: local`, which makes
+ *     their position fixed to the _content_ within the scrollbox.
+ *   - One gradient each at the top and the bottom of the frame that are the
+ *     shadow hints (shadows). These use `background-attachment: scroll` such
+ *     that they are always positioned at the top and the bottom of the
+ *     _scrollbox_ frame. When these positions align with the positions of the
+ *     shadow covers--at the top and the bottom of the overflowing content--
+ *     they will be obscured by those shadow covers.
+ *
+ * See https://lea.verou.me/2012/04/background-attachment-local/
+ *
+ * Safari's behavior is different because of a bug with
+ * `background-attachment: local`.
+ * See https://bugs.webkit.org/show_bug.cgi?id=219324
```

Comment:
This bug is now marked as fixed. Reading through the comments it looks like this issue affects Safari 14 through to Safari 15.2 (Safari 15.3 was released after the fix landed, though I don't know for sure that it includes this change). Everything seems to work correctly in my testing in Safari 15.4. Do you want to check and confirm?

If so then we would note that only certain Safari versions are affected.

Reply:
Ah, still broken in 15.2. I need to perform a system update to update my Safari to test in 15.4. I can do that later today, but don't want to take the time to do the update at this moment.

To wit, I see only bottom shadows (and they never go away) in Safari 15.2.

Reply:
Comment updated.